chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -612,70 +606,103 @@ impl NetworkState {
         Ok(edge)
     }
 
-    /// Check for edges indicating that
+    pub async fn add_accounts_data(
+        self: &Arc<Self>,
+        accounts_data: Vec<Arc<SignedAccountData>>,
+    ) -> Option<accounts_data::Error> {
+        let this = self.clone();
+        self.spawn(async move {
```

Comment:
Unrelated to the PR, but I wonder if `spawn` should create a _new_ spawn, so that we can easily observe how long each of netwrok_state methods work

Reply:
you mean new span?

Reply:
Sorry, something happend with grammar there. Yeah, I meant that we should maybe do something like

```
self.spawn("add_accounts_data", async move {)
```

and have that `spawn` create an extra `span`. "Which methonds of network state are slow/bottleneck" feels like an important data to observe